### PR TITLE
[exporter/datadog]: ensure top level spans are computed

### PR DESCRIPTION
**Pull request by ericmustin**
**Description:** <Describe what has changed.>

A previous PR accidentally dropped top level span computation, (here: https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/269678310105311d8eb652d60f66fdd66eceb396#diff-b98fa670a9a811a13714fe4ce9bb4b3c5c6442c05638eb156a61e5480e8aa8faL82) which has caused an unintended effect of causing a large number of spans to not be retained. This PR reverts that change

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

updated unit tests

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]
Thanks for the quick fix :)

File: exporter/datadogexporter/translate_traces.go [hunk 1]
```diff
@@ -208,6 +209,9 @@ func resourceSpansToDatadogSpans(rs pdata.ResourceSpans, hostname string, cfg *c
 		// TODO: allow users to configure specific spans to be marked as an analyzed spans for app analytics
 		top := getAnalyzedSpans(apiTrace.Spans)
 
+		// https://github.com/DataDog/datadog-agent/blob/443930a00bced1ea87d0067fc2f1fc3f98a48ba1/pkg/trace/agent/agent.go#L241-L243
+		traceutil.ComputeTopLevel(apiTrace.Spans)
```
Comment (mx-psi) [inline exporter/datadogexporter/translate_traces.go:213]: Is there any unit test we can add for this to catch it?

Comment (ericmustin) [reply exporter/datadogexporter/translate_traces.go:213]: well, i added back the tests that got removed here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3786/files#diff-0150f0fe9ebf825e7a4cebf5bdc61d443f910ae89c8b512bb4bc5f2c57dc7b56R803 . That gets us some of the way there.

I think I just missed the importance of this metric tbh

File: exporter/datadogexporter/stats.go [hunk 2]
```diff
@@ -37,6 +37,7 @@ func computeAPMStats(tracePayload *pb.TracePayload, pushTime int64) *stats.Paylo
 	bucketTS := pushTime - statsBucketDuration
 	for _, trace := range tracePayload.Traces {
 		spans := getAnalyzedSpans(trace.Spans)
+
```
Comment (bogdandrutu) [inline exporter/datadogexporter/stats.go:40]: Unnecessary change.

**Review by bogdandrutu — APPROVED** [review]
